docs/user-tutorial/data-diagnosis.md

```diff
@@ -0,0 +1,144 @@
+
+    `Tips`: this file for the partly representative machine types will be published in [SuperBench Results Repo](https://github.com/microsoft/superbench-results/tree/main) with the release of Superbench.
+
+### Rule file
+
+This section describes how to write rules in **Rule file**.
+
+The convention is the same with [SuperBench Config File](https://microsoft.github.io/superbenchmark/docs/superbench-config), please view it first.
+
+Here is an overview of the Rule file structure:
+
+scheme:
+```yaml
+version: string
+superbench:
+  var:
+    ${var_name}: dict
+  rules:
+    ${rule_name}:
+      function: string
+      criteria: string
+      categories: string
+      metrics:
+        - ${benchmark_name}/regex
+        - ${benchmark_name}/regex
+        ...
+```
+
+example:
+```yaml
+# SuperBench rules
+version: v0.3
+superbench:
+  rules:
+    failure-rule:
+      function: value
+      criteria: '>,0'
+      categories: Failed
+      metrics:
+        - kernel-launch/return_code
+        - mem-bw/return_code
+        - nccl-bw/return_code
+        - ib-loopback/return_code
+    rule0:
+    # Rule 0: If KernelLaunch suffers > 5% downgrade, label it as defective
+      function: variance
+      criteria: '>,5%'
+      categories: KernelLaunch
+      metrics:
+        - kernel-launch/event_overhead:\d+
+        - kernel-launch/wall_overhead:\d+
+    rule1:
+    # Rule 1: If H2D_Mem_BW or D2H_Mem_BW test suffers > 5% downgrade, label it as defective
+      function: variance
+      criteria: '<,-5%'
+      categories: Mem
+      metrics:
+        - mem-bw/H2D_Mem_BW:\d+
+        - mem-bw/D2H_Mem_BW:\d+
+    rule2:
+    # Rule 2: If NCCL_BW suffers > 5% downgrade, label it as defective
+      function: variance
+      criteria: '<,-5%'
+      categories: NCCL
+      metircs:
+        - nccl-bw/allreduce_8589934592_busbw:0
+    rule3:
+    # Rule 3: If GPT-2, BERT suffers > 5% downgrade, label it as defective
+      function: variance
+      criteria: '<,-5%'
+      categories: Model
+      metrics:
+        - bert_models/pytorch-bert-base/throughput_train_float(32|16)
+        - bert_models/pytorch-bert-large/throughput_train_float(32|16)
+        - gpt_models/pytorch-gpt-large/throughput_train_float(32|16)
+```
+
+This Rule file describes the rules used for data diagnosis.
+
+They are firstly organized by the rule name, and each rule mainly includes 4 elements:
+
+#### `metrics`
+
+The list of metrics for this rule. Each metric is in the format of ${benchmark_name}/regex, you can use regex after the first '/', but to be noticed, the benchmark name can not be a regex.
+
+#### `categories`
+
+The categories belong to this rule.
+
+#### `criteria`
+
+The criteria used for this rule, which indicate how to compare the data with the baseline value. The format is `${operator},${value}`.
+
+- ${operator}: comparison operator, < or >
+- ${value}: the threshold, float or percentage
```

Comment:
The criteria can be a standard formula?

Reply:
The criteria can be a standard formula, but do you think '>5%' is a standard formula, it can not be directly calculated by eval(1>5%). 
If I define that the criteria must be directly calculated by eval() and output as bool, does it work for you?

Reply:
How about '>0.05'? What I thought is that maybe we can support more complex criteria without code change in the future.

Reply:
Yes, I agree with Guoshuai. Can we support a lambda function directly?

Reply:
sure, good idea, changed to lambda function